Voice determination (active or passive) functionality for Dutch sentences

ThijsVroegh opened the pull request:
This pull request includes several changes to the `orangecontrib/storynavigation/modules/tagging.py` file, focusing on adding voice determination functionality for Dutch sentences and improving code readability. The most important changes include updating the data columns, adding a new method to process Dutch voice, and appending the voice information to the data rows. 

Solves issue #126 

### Voice determination and data processing:

* Updated `self.complete_data_columns` to include the new 'voice' column.
* Added a new method `__process_dutch_voice` to determine if a Dutch sentence or clause is in the passive or active voice.
* Modified `__parse_tagged_story` to determine the voice for each sentence and append this information to the data rows. [[1]](diffhunk://#diff-fb61e723cc0c19cb0c99a7dd27f31e4435001acfdf025c2c3861018f7199f3c4R157-R184) [[2]](diffhunk://#diff-fb61e723cc0c19cb0c99a7dd27f31e4435001acfdf025c2c3861018f7199f3c4R194)

### Code readability improvements:

* Improved the docstring format for the `__parse_tagged_story` method for better readability.
* Added whitespace for better code organization and readability.

eriktks reviewed (approved):
I have examined the code and tested it by running it. The code adds an extra column "voice" to the output table of the elements widget which states whether a sentence is in active or passive voice. The code does not work perfectly: it is expected that "moeten worden gezien" would be classified as passive while "moeten gezien worden" would be classified as active. Using the dependency parsing analysis of Spacy could help here. However this can be left as future work. Thanks for this solution!

ThijsVroegh commented:
Thanks @eriktks!

I changed the code so that both "moeten worden gezien" and "moeten gezien worden" would be classified as passive, as should be the case. Calling the function runs properly, but perhaps the function itself should be further optimized based on the rules and exceptions for active or passive classification. In addition to 'worden', 'zijn' is also a verb used to make a sentence passive. This is a little bit more complicated though, because 'zijn' as auxiliary verb is not always automatically leading to a passive sentence, whereas 'worden' does. Anyway, I made the function more flexible to handle more specific cases:

 - Enhanced the function __process_dutch_voice to determine if a Dutch sentence is in the passive or active voice.
-  Introduced new checks and flags to better detect passive voice using auxiliary verbs and past participles.
-  Improved the handling of prepositional agents and syntactic heads for context.
-  Added detailed comments and removed redundant code to streamline the function.

Perhaps we should get more clear 'rules' from the LA to decide when something is active or passive?

eriktks commented on orangecontrib/storynavigation/modules/tagging.py:
```diff
@@ -326,43 +326,118 @@ def __process_dutch_future_verbs(self, tags, tag):
+            pos_value = tok[1]  
+            tag_value = tok[2]  
+            dep_value = tok[3]  # Dependency relation
+            head_value = tok[5].head.text  # Syntactic head (index of the parent token)
```
```suggestion
            head_value = tok[5].head.lemma_
```
Use "lemma_" here, like in the rest of the code

eriktks reviewed (changes requested):
Thanks for these updates! I have tested and checked the code and made two change suggestions. Otherwise the code worked well and looks fine.
